src/components/ArcGISWebMap.tsx

```diff
@@ -159,14 +169,138 @@ function FormElementArcGISWebMap({
+          }),
+      )
+    } else if (geom.type === 'polyline') {
+      const polyline = geom as Polyline
+      return polyline.paths[0].map(
+        (path) =>
+          new Point({
+            x: path[0],
+            y: path[1],
+            spatialReference: spatialRef,
+          }),
+      )
+    }
+  }
+
+  const addMeasurementLabels = React.useCallback(
+    (graphics: __esri.Graphic[]) => {
+      const spatialRef = new SpatialReference({ wkid: 3857 })
+      const measurementLayer = measurementLayerRef.current
+      const mapView = mapViewRef.current
+      if (!measurementLayer || !mapView) return
+      measurementLayer.removeAll()
+
+      for (const graphic of graphics) {
+        const geom = graphic.geometry
+        if (!geom || (geom.type !== 'polygon' && geom.type !== 'polyline'))
+          continue
+
+        const points = getGeometryPoints(geom, spatialRef)
+        if (!points) continue
+
+        const { distanceUnit, distanceUnitShortName } =
+          localisationService.getDistanceUnits()
+        if (distanceUnit !== 'meters' && distanceUnit !== 'feet') {
+          console.warn(
+            'Unsupported distance unit provided in tenant configuration: ',
+            distanceUnit,
+          )
+          return
+        }
+        const graphics: Graphic[] = []
+
+        for (let i = 1; i < points.length; i++) {
+          const distance = geometryEngine.distance(
+            points[i],
+            points[i - 1],
+            distanceUnit,
+          )
+
+          const x1 = points[i].x
+          const x2 = points[i - 1].x
+          const y1 = points[i].y
+          const y2 = points[i - 1].y
+
+          const midpoint = new Point({
+            x: (x1 + x2) / 2,
+            y: (y1 + y2) / 2,
+            spatialReference: spatialRef,
+          })
+
+          // Angle of the measurement label - this will be rotated to be parallel with the polygon or polyline edge
+          const dx = x2 - x1
+          const dy = y2 - y1
+          const radians = Math.atan2(dy, dx)
+          let degrees = radians * (180 / Math.PI)
+          if (degrees > 90 || degrees < -90) {
+            degrees += 180
+          }
+
+          const pixelOffset = 10
+          const normalAngle = radians + Math.PI / 2
+          const offsetScreenX = pixelOffset * Math.cos(normalAngle)
+          const offsetScreenY = pixelOffset * Math.sin(normalAngle)
+
+          const screenPoint = mapView.toScreen(midpoint)
+          screenPoint.x += offsetScreenX
+          screenPoint.y -= offsetScreenY
+          const offsetMapPoint = mapView.toMap(screenPoint)
+
+          graphics.push(
+            new Graphic({
+              geometry: offsetMapPoint,
+              symbol: new TextSymbol({
+                text: distance.toFixed(0) + distanceUnitShortName,
+                color: 'black',
+                haloColor: 'white',
+                haloSize: 1,
+                angle: -degrees,
```

Comment:
might be worthwhile putting the - in the calculation and updating the 
`if (degrees > 90 || degrees < -90) {
            degrees += 180
          }` logic? Maybe rename var to `angle` instead of degrees?